### ISSUE TITLE
FSSpec dependency

### ISSUE THREAD

**Issue by mrocklin** [dask/dask#5119]
A few quesitons about the recent fsspec dependency:

1.  Do we now want to list this as a hard dependency?  It's currently listed in our setup.py under `install_requires=`, and yet I think in other places we say that core Dask has no dependencies beyond the standard library.
2.  Where should it live?  Should it be in the Dask org so that maintainers of this project have control over it?
3.  How should we handle downstream project CI systems?  Currently PRs in at least dask/distributed are failing.

**Comment by TomAugspurger** [comment]
If possible, it'd be nice to keep it an optional dependency.

We'll have the the Dask Conda package depend on it directly though. And the
"dataframe" extras pip package will as well.

If it's optional, we should probably have runtime logic checking the
version of fsspec / s3fs, to ensure users get the right versions.

**Comment by martindurant** [comment]
I didn't spot this (thought we decided in the last meeting).

Dask's bytes stuff was, obviously, not optional. I *suppose* you don't actually reach of any of that stuff if you don't use IO functions in bag, array or dataframe, so the dep could be removed from dask_core with a little work.

The final versions to require can be decided at the time of the next release, but there is some pressure, as users upgrading s3fs or gcsfs will see failures until then. Releasing those was required to get tests to pass, so some chicken/egg going on. As far as distributed is concerned, the easiest remedy is simply to include fsspec in the build (this can be temporary).

**Comment by martindurant** [comment]
NB: `import dask` does not import fsspec even now.

**Comment by TomAugspurger** [comment]
Does distributed need a hard dependency on fsspec?

**Comment by martindurant** [comment]
I assume the tests are using non-core functions, so for the builds, yes. For dask-core only, I don't think so.

**Comment by mrocklin** [comment]
Yeah, for context, Dask core doesn't even strictly depend on `toolz`, which is used broadly throughout the codebase.  I think that we should remove the `fsspec` dependency.

**Comment by mrocklin** [comment]
I think that we also need to get fsspec into the Anaconda `defaults` channel quickly before Dask can be released.

**Comment by martindurant** [comment]
https://github.com/dask/dask/pull/5121

**Comment by martindurant** [comment]
https://github.com/dask/distributed/pull/2859
